### PR TITLE
filter items

### PR DESCRIPTION
**Pull request by amiryehu**
create url by filters
And some touches of design

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -4,12 +4,19 @@ import { StyledButton } from "./Button-style";
...
 
 const Button = (props: IButton) => {
-  const {className, btnText} = props;
-  return <StyledButton className={className}>{btnText}</StyledButton>;
+  const {className, btnText, articleUrl} = props;
+
+  const openArticleInNewTab = (): void => {
+    const newWindow = window.open(articleUrl, '_blank', 'noopener,noreferrer')
+    if (newWindow) newWindow.opener = null
+  }
```
Comment (ofekbd-moveo) [inline src/components/Button/Button.tsx]: Better to pass that as onClickHandler function

File: src/components/CardContainer/CardContainer.tsx [hunk 2]
```diff
@@ -2,35 +2,66 @@ import React, { useEffect } from "react";
...
+        const currentLocation = response.country_code.toLowerCase();
+
+        dispatch(filtersActions.setCountry(currentLocation));
+        dispatch(filtersActions.setCountryName(countryName));
+      }
+    })
+    .catch((data) => {
+      console.log('Request failed:', data);
+    });
+  },[filtersState.Endpoint])
```
Comment (ofekbd-moveo) [inline src/components/CardContainer/CardContainer.tsx]: Move this to MainContent component

File: src/components/CardContainer/CardContainer.tsx [hunk 3]
```diff
@@ -2,35 +2,66 @@ import React, { useEffect } from "react";
...
+    }
+    dispatch(fetchFromApi(url));
+  }, [
+    filtersState.Endpoint,
+    filtersState.Category,
+    filtersState.Country,
+    filtersState.Sources,
+    filtersState.Language,
+    filtersState.SortBy,
+    filtersState.valueInSearch
```
Comment (ofekbd-moveo) [inline src/components/CardContainer/CardContainer.tsx:34]: Maybe work with just filtersState too. check it

File: src/components/Filter/Filter.tsx [hunk 4]
```diff
@@ -17,26 +19,36 @@ interface IFilter {
...
   const IsEndPointOrRegularFilter = (isEndpoint:boolean | undefined):string => {
     if (isEndpoint) return Endpoint.TopHeadlines;
     return "";
   }
 
-  const [selectedItem, setSelectedItem] = useState(IsEndPointOrRegularFilter(isEndpointFilter));
+  useEffect(()=>{
+    if (filterState[title as TitlesOptions] === ""){
+      setSelectedItem(()=>"")
+    }
```
Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx]: To make your code more clearable - you can get it out to function or add comment - "reset selected value" / "clean selected value"

Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx]: To make your code more clearable, you can get it out to a function or add comment - "reset selected value" / "cleanup selected value"

Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx]: To make your code more clearable - you can get it out to a function or add comment -"reset selected value" / "cleanup selected value".

File: src/components/Filter/Filter.tsx [hunk 5]
```diff
@@ -17,26 +19,36 @@ interface IFilter {
...
+  },[filterState[title as TitlesOptions]])
+
+  const [selectedItem, setSelectedItem] = useState(()=>IsEndPointOrRegularFilter(isEndpointFilter));
```
Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx]: Usually functions that start with "is..." return boolean value. Here you initialize selected item value so you can call that function "initByFilterType" or something like that..

File: src/components/Filter/Filter.tsx [hunk 6]
```diff
@@ -47,46 +59,51 @@ export default function BasicSelect(props: IFilter) {
...
-            sx={{ ...selectStyle }}
             value={selectedItem}
+            onChange={handleChange} 
+            sx={{ ...selectStyle }}
             renderValue={(value) => {
-              if (value.length === 0) {
+              if (value === "") {
                 return <em>{title}</em>;
               } else {
+                console.log(selectedItem)
```
Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx]: delete

File: src/components/Search/Search.tsx [hunk 7]
```diff
@@ -7,20 +7,34 @@ import {  SearchWrapper, FilterDividerWrapper, SearchContainer, textFieldStyle,
...
+import { filtersActions } from "../../store/reducers/filterReducer";
+
 
 const SearchForm = () => {
+  const dispatch = useAppDispatch();
   const [isFocused, setIsFocused] = useState(false);
   const handleFocusState = () => {setIsFocused((state)=> !state)}
+  const valueInSearch = useAppSelector(state=>state.filters.valueInSearch);
+  
+  const valueCangeInSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (ofekbd-moveo) [inline src/components/Search/Search.tsx]: *valueChangeInSearch
Usually, event handlers functions start with onEventHandler. 
onSearchChangeHandler.

File: src/store/Utils/filtersData.ts [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+  United_States: "us",
+  Venuzuela: "ve",
+};
+export type CountriesOptions = "Argentina"| "Australia"| "Austria"| "Belgium"| "Brazil"| "Bulgaria"| "Canada"| "China"| "Colombia"| "Cuba"| "Czech_Republic"| "Egypt"| "France"| 
+                                "Germany"| "Greece"| "Hong_Kong"| "Hungary"| "India"| "Indonesia"| "Ireland"| "Israel"| "Italy"| "Japan"| "Latvia"| "Lithuania"| "Malaysia"| "Mexico"| 
+                                "Morocco"| "Netherlands"| "New_Zealand"| "Nigeria"| "Norway"| "Philippines"| "Poland"| "Portugal"| "Romania"| "Russia"| "Saudi_Arabia"| "Serbia"| 
+                                "Singapore"| "Slovakia"| "Slovenia"| "South_Africa"| "South_Korea"| "Sweden"| "Switzerland"| "Taiwan"| "Thailand"| "Turkey"| "UAE"| "Ukraine"| 
+                                "United_Kingdom"| "United_States"| "Venuzuela"; 
+                                
+// export const CountriesObjArray = [
```
Comment (ofekbd-moveo) [inline src/store/Utils/filtersData.ts]: delete

File: src/store/Utils/filtersData.ts [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    Reuters: "Reuters",
+    Slashdot_org: "Slashdot.org",
+    TechCrunch: "TechCrunch",
+    The_Guardian: "The Guardian",
+    The_Next_Web: "The Next Web",
+    The_Verge: "The Verge",
+    Wired: "Wired",
+};
+export type SourcesOptions = "Android_Central"| "Ars_Technica"| "Business_Insider"| "Engadget"| "Entrepreneur"| "Gizmodo_com"| "MacRumors"| "MarketBeat"| 
+                             "New_York_Times"| "Reuters"| "Slashdot_org"| "TechCrunch"| "The_Guardian"| "The_Next_Web"| "The_Verge"| "Wired";
```
Comment (ofekbd-moveo) [inline src/store/Utils/filtersData.ts:114]: I think you should execute another api call to get all sources

File: src/store/Utils/index.ts [hunk 10]
```diff
@@ -18,4 +21,32 @@ export const fetchFromApi = createAsyncThunk("fetchFromApi",  async (url: string
...
+
+
+// export const getUserLocation = () => {
+//      fetch("https://ipapi.co/json/")
+//     (res: any)=>{
+//       const objLocation =   res.json();
+//       return objLocation.country_code.toLowerCase();
+//     }
+//     fetchFromApiErrorHandler();
+// };
```
Comment (ofekbd-moveo) [inline src/store/Utils/index.ts]: delete

File: src/components/CardContainer/CardContainer.tsx (hunk 3 above)
Comment (amiryehu) [reply src/components/CardContainer/CardContainer.tsx:34]: I have a variable called "searchInFocus" so I can coordinate between "search" and "recentSearches".
I need to listen to everyone except him

**Review by ofekbd-moveo — APPROVED** [review]

**Pull request merged**